doc/source/deploy-on-kubernetes.rst

```diff
@@ -1,148 +1,293 @@
+  `Questions or Issues?`_ section below.
 
-You can run Ray on top of Kubernetes. This document assumes that you have access
-to a Kubernetes cluster and have ``kubectl`` installed locally.
+This document assumes that you have access to a Kubernetes cluster and have
+``kubectl`` installed locally and configured to access the cluster.
 
-Start by cloning the Ray repository.
+Creating a Ray Namespace
+------------------------
+
+First, create a `Kubernetes Namespace`_ for Ray resources on your cluster. The
+following commands will create resources under this Namespace, so if you want
+to use a different one than ``ray``, please be sure to also change the
+`namespace` fields in the provided ``yaml`` files and anytime you see a ``-n``
+flag passed to ``kubectl``.
 
 .. code-block:: shell
 
-  git clone https://github.com/ray-project/ray.git
+  $ kubectl create -f ray/doc/kubernetes/ray-namespace.yaml
 
-Work Interactively on the Cluster
----------------------------------
+Starting a Ray Cluster
+----------------------
 
-To work interactively, first start Ray on Kubernetes.
+A Ray cluster consists of a single head node and a set of worker nodes (the
+provided ``ray-cluster.yaml`` file will start 3 worker nodes). In the example
+Kubernetes configuration, this is implemented as:
 
-.. code-block:: shell
+- A ``ray-head`` `Kubernetes Service`_ that enables the worker nodes to discover the head node when they start up.
+- A ``ray-head`` `Kubernetes Deployment`_ that backs the ``ray-head`` Service with a single head node pod (replica).
+- A ``ray-worker`` `Kubernetes Deployment`_ with multiple worker node pods (replicas) that connect to the ``ray-head`` pod using the ``ray-head`` Service..
 
-  kubectl create -f ray/kubernetes/head.yaml
-  kubectl create -f ray/kubernetes/worker.yaml
+Note that because the head and worker nodes are Deployments, Kubernetes will
+automatically restart pods that crash to maintain the correct number of
+replicas.
 
-This will start one head pod and 3 worker pods. You can check that the pods are
-running by running ``kubectl get pods -n ray``.
+- If a worker node goes down, a replacement pod will be started and it should join the cluster normally.
+- If the head node goes down, it will be restarted and effectively start a new cluster. Worker nodes that were connected to the old head node will crash due to losing connection to the original head node, and will subsequently be restarted and connect to the new head node.
 
-You should see something like the following (you will have to wait a couple
-minutes for the pods to enter the "Running" state).
+Try deploying a cluster with the provided Kubernetes config by running the
+following command:
 
 .. code-block:: shell
 
-  $ kubectl get pods -n ray
+  $ kubectl apply -f ray/doc/kubernetes/ray-cluster.yaml
+
+Vrify that the pods are running by running ``kubectl get pods -n ray``. You
+should see something like the following (you may have to wait up to a few
```

Comment:
nit: this line is redundant; if you just show the output below I think it will be implied.